backend/agent_logic/orchestrator.py

```
@@ -40,27 +43,44 @@ async def validate_api_key(self, api_key: str) -> bool:
-            cache[task_id]["status"] = "completed"
-            logger.info(f"Task {task_id} completed successfully")
+            if result["success"]:
+                cache[task_id]["status"] = "completed"
+                cache[task_id]["logs"].append("? Task completed successfully!")
+                logger.info(f"? Task {task_id} completed successfully")
+            else:
+                cache[task_id]["status"] = "error"
+                cache[task_id]["error"] = result.get("error", "Unknown error")
+                logger.error(f"? Task {task_id} failed: {result.get('error')}")
             
         except Exception as e:
```

Ruff (0.14.2)

Do not catch blind exception: `Exception`

(BLE001)